docs/01-tooling/02-better-minification-for-frontend-sources.md

```diff
@@ -1,6 +1,87 @@
 ## Better minification for the frontend sources
+
 ### Intro
+
+MediaWiki via [ResourceLoader][rl-min] uses [JavaScriptMinifier][] to minimize
+JavaScript files so that their size is as small as possible.
+
+Since the minification happens in the PHP server at runtime, tradeoffs were made
+so that the server could minify in a performant way, giving up size gains for
+speed.
+
+The JavaScript ecosystem has continued evolving minifiers based on Node.js
+tooling that can't be used on a PHP server easily.
+
+There are gains to be had if we were to use node based minifiers and just give
```

Comment:
Again, _Node.js_.

Reply:
The finishing point here can be tied back to the RL's cache: the simpler the dependency tree, the easier it is for the RL to decide if a module has been invalidated.

Reply:
Wouldn't that point be better suited for the bundling post?

Reply:
It is a good point, so I've added it to the first post, see b1f4b8a